From f79b5a04a705706cce301b92556a4f7e8ad932c8 Mon Sep 17 00:00:00 2001
From: Chris Coulson <chrisccoulson@googlemail.com>
Date: Sun, 4 Oct 2009 20:50:25 +0100
Subject: [PATCH] Set status icon invisible before unref'ing it

Set the status icon invisible before unref'ing it. This avoids
leaving whitespace in the notification area, which grows in size
each time the icon is displayed/hidden
---
 src/main.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/main.c b/src/main.c
index 5970670..a8970df 100644
--- a/src/main.c
+++ b/src/main.c
@@ -141,6 +141,7 @@ update_temporary_authorization_icon_real (void)
       /* hide icon */
       if (status_icon != NULL)
         {
+          gtk_status_icon_set_visible (status_icon, FALSE);
           g_object_unref (status_icon);
           status_icon = NULL;
         }
-- 
1.6.3.3

